src/foam/core/XMLSupport.java

```diff
@@ -0,0 +1,158 @@
+
+    // TODO: Planning to add support for other date formats in the future
+
+    // Parsing xml for FObjects and inserting into provided XMLDAO
+    public static void XMLToFObject (String fileName, XMLDAO xmlDAO) throws FileNotFoundException, XMLStreamException, IllegalAccessException {
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+        try {
+            XMLStreamReader xmlr = xmlInputFactory.createXMLStreamReader(new FileReader(fileName));
+            int eventType;
+            Object clsInstance = null;
+            while (xmlr.hasNext()){
+                eventType = xmlr.next();
+                switch (eventType) {
+                    case XMLStreamConstants.START_DOCUMENT:
+                        break;
+                    case XMLStreamConstants.START_ELEMENT:
+                        if (xmlr.getLocalName().equals("object")){
+                            // Create new fObject
+                            String objClass = xmlr.getAttributeValue(null, "class");
+                            Class<?> cls;
+
+                            try {
+                                cls = Class.forName(objClass);
+                                clsInstance = cls.newInstance();
+                                ClassInfo csInfo = ((FObject)clsInstance).getClassInfo();
+                                xmlDAO.setOf(csInfo);
+                            } catch (ClassNotFoundException | InstantiationException ex) {
+
+                            }
+                            // Object properties
+                            fillProperties((FObject) clsInstance, xmlr);
+                            if (clsInstance != null){
+                                xmlDAO.putOnly( (FObject) clsInstance);
+                            }
+                        }
+                        break;
+                    case XMLStreamConstants.END_ELEMENT:
+                        break;
+                    case XMLStreamConstants.END_DOCUMENT:
+                        break;
+                }
+            }
+            xmlr.close();
+        } catch (FileNotFoundException ex) {
+            throw new FileNotFoundException("Xml file not found with given filename: " + fileName);
+        } catch (XMLStreamException ex) {
+            throw new XMLStreamException("Premature end of xml file");
+        }
+    }
+
+    private static void fillProperties(FObject obj, XMLStreamReader reader) throws XMLStreamException {
+        try {
+            int eventType;
+            PropertyInfo prop = null;
+            while (reader.hasNext()) {
+                eventType = reader.next();
+                switch (eventType){
+                    case XMLStreamConstants.START_ELEMENT:
+                        prop = (PropertyInfo) obj.getClassInfo().getAxiomByName(reader.getLocalName());
+                        if (prop == null) {
+                            throw new XMLStreamException("Could not find property " + reader.getLocalName());
```

Comment:
I don't think this should be an exception. Just log at most. If I remove properties from a model I still want to be able to load old XMLDAO files and have it just ignore the removed properties, not refuse to load the data and leave me broken.